### PR TITLE
Make flutter and dart scripts invoke their batch file equivalents on Windows

### PR DESCRIPTION
**Pull request by gspencergoog**
## Description

This makes the `flutter` and `dart` scripts invoke their batch file equivalents if running under MINGW (i.e. git-bash) on Windows.

This allows for proper locking, and makes sure that people aren't using two different (and non-mutally-aware) locking systems when running flutter on Windows.

I also fixed a couple of places where we look for MINGW32, which fails under MINGW64. It just looks for MINGW now.

## Related Issues

 - https://github.com/flutter/flutter/issues/58993

### CODE REVIEW

File: bin/dart [hunk 1]
```diff
@@ -43,8 +43,14 @@ function follow_links() (
...
 )
 
-PROG_NAME="$(follow_links "$BASH_SOURCE")"
+PROG_NAME="$(follow_links "${BASH_SOURCE[0]}")"
 BIN_DIR="$(cd "${PROG_NAME%/*}" ; pwd -P)"
+OS="$(uname -s)"
+
+if [[ $OS =~ MINGW.* ]]; then
+  # If we're on Windows, invoke the batch script instead, to get proper locking.
+  exec "${BIN_DIR}/dart.bat" "$@"
```
Comment (christopherfujino) [inline bin/dart:52]: do we want to exit after this?

Comment (gspencergoog) [reply bin/dart:52]: The [`exec` command](https://www.computerhope.com/unix/bash/exec.htm) completely substitutes the given command for the current process, so there's no need to exit (it already has exited by invoking the other script).

Comment (gspencergoog) [reply bin/dart:52]: Also, the final return value is the return value of the other script.

Comment (christopherfujino) [reply bin/dart:52]: cool

Comment (christopherfujino) [reply bin/dart:52]: I guess that means we should never reach the other switch cases on MINGW anyway?

Comment (gspencergoog) [reply bin/dart:52]: What other switch cases?  I mean, no, it'll never even reach the "fi" because it execs the other process immediately.

**Review by christopherfujino — APPROVED** [review]
LGTM

**Pull request merged**